Controllers/ProductTypesController.cs

```diff
@@ -28,16 +28,21 @@ public ProductTypesController(BangazonContext ctx)
             context = ctx;
         }
 
-        public List<ProductType> Buy()
-        { 
-            return context.ProductType.ToList();
-            //return View(await context.ProductType.ToListAsync()); 
+        public async Task<IActionResult> Buy()
+        {
+            List<ProductType> ProductTypeList = await context.ProductType.ToListAsync();
+            ProductTypeList.ForEach(CalculateTypeQuantities);
+            return View(ProductTypeList);
         }
-        public List<ProductType> Sell()
+        public async Task<IActionResult> Sell()
         {
-            return context.ProductType.ToList();
-            //return View(await context.ProductType.ToListAsync()); 
+            return View(await context.ProductType.ToListAsync()); 
         }
 
+        public void CalculateTypeQuantities(ProductType productType)
```

Comment:
Something to consider is moving `CalculateTypeQuantities` into an [extension method](https://msdn.microsoft.com/en-us//library/bb383977.aspx) so we can reuse this function in other places in our code in the future. This also reduces the need to keep a `Quanitity` property on the model since `Quanity` would have to be updated every time a new product is created/destroyed.